lib/modules/platform/gitea/pr-cache.ts

```diff
@@ -134,9 +134,21 @@ export class GiteaPrCache {
         break;
       }
 
-      url = parseLinkHeader(res.headers.link)?.next?.url;
+      if (process.env.RENOVATE_X_REBASE_PAGINATION_LINKS) {
```

Comment:
it seems this isn't the only place for gitea pagination. check util/http folder

Reply:
As far as I can tell this is the only place that has the problem. The pagination in util/http/gitea.ts doesn't rely on the url that is returned by gitea but uses the baseUrl for every request instead (line 54). 
I could see a possible problem with this behavior because this means that two different ways are used to paginate, but i don't think that I should fix it in this PR because it was like this before my changes.